### ISSUE TITLE
matplotlib rc('text', usetex=True) breaks plotting from .ipynb to .py (jupyter --> spyder)

### ISSUE THREAD

**Issue by louika** [spyder-ide/spyder#3606]
## Description of your problem

I loaded a .py file from jupyter to use it in spyder. But the plotting does not work anymore. I narrowed it down to the rc('text', usetex=True) command. I asked this on stackoverflow:

Original Question:
http://stackoverflow.com/questions/40266888/migrating-jupyter-notebook-to-py-file-spyder-basic-plotting-is-killed-by-late
## What steps will reproduce the problem?

create a .py script in spyder:

```
import numpy as np
import matplotlib as mpl
import matplotlib.pyplot as plt
import matplotlib.gridspec as gs

from matplotlib import rc # this is the matplotlib suggestion
rc('font',**{'family':'sans-serif','sans-serif':['Arial']})
rc('text', usetex=True)

mat = np.random.rand(5,5)

fig = plt.figure()
gsall = gs.GridSpec(6, 1)
ax1 = fig.add_subplot(gsall[:])
source1 = ax1.contourf(mat,extend='both',cmap=mpl.cm.viridis)
plt.title('Random field \n test')
plt.colorbar(source1)
plt.show()
```

run it.
## What is the expected output? What do you see instead?

I expect a figure with the random matrix and the text, but I get many errors.
## Please provide any additional information below

In jupyter notebook everything is fine. In ipython shell I get a figure without the text and no errors.
## Versions and main components
- Spyder Version: 2.3.9
- Python Version: 3.5.1 
- Operating system: macOS sierra v.10.12.1
- Matplotlib Version: 1.5.1

**Comment by ccordoba12** [comment]
Could you update to Spyder 3.0 and try again? Spyder 2.3.9 is no longer maintained :-)

**Comment by louika** [comment]
thanks @ccordoba12 the scripts are working now. But Anaconda Navigator does not start anymore. I can start jupyter and spyder from the command line, but the navigator is very useful for me. So I will try to find a solution for that.

**Comment by malliwi88** [comment]
Hi @louika 

I think you only need:

* Latest version of spyder (3.0.1)
* sudo apt-get install dvipng (library for manage latex character conversion on matplotlib)

for you can to reproduce [this example](http://matplotlib.org/users/usetex.html) without problems over Ipython console.

![seleccion_035](https://cloud.githubusercontent.com/assets/5307894/19808149/ef146a92-9ce8-11e6-9fbc-485c9b51e36f.png)


I hope this help you.

**Comment by ccordoba12** [comment]
Closing, as the last line of the error you posted in StackOverflow

    FileNotFoundError: [Errno 2] No such file or directory: 'kpsewhich'

means that you don't have Latex installed, or that is not configured correctly.

**Comment by texadactyl** [comment]
And, this has nothing to do with Spyder.
I get the same headache on the command line.
Seems to be a Tkinter / latex issue.

Sacre merde alors!  (=:

**Comment by ccordoba12** [comment]
> Yes, I have latex installed with pip and pip3

What do you mean by this? Latex can't be installed with pip, you need to install with your package manager (i.e. apt-get, emerge, etc, since you're using Linux).

**Comment by texadactyl** [comment]
`sudo pip3 install latex`
installs a Python package called "latex".

However, there must be, as you said, a native-code executable?  Okay.
You use a debian-ish Linux? If so, which package gets the latex executable?
If RPM, then the same question.

Thank you.

**Comment by texadactyl** [comment]
Debian/Ubuntu/Mint package texlive-full seems to be an effective solution (yuge!).  Also can be downloaded from http://www.tug.org/texlive/

Anyways, as I said before, this has nothing to do with spyder.

**Comment by texadactyl** [comment]
Debian/Ubuntu/Mint package texlive-full seems to be an effective solution (yuge!).  Also can be downloaded from http://www.tug.org/texlive/

Anyways, as I said before, this has nothing to do with spyder.

**Comment by ImportanceOfBeingErnest** [comment]
[This comment](https://stackoverflow.com/questions/50886030/matplotlib-rctext-usetex-true-producing-errors#comment88786536_50886030) suggests that it may still be related to spyder after all.

**Comment by ccordoba12** [comment]
@ImportanceOfBeingErnest, if you define or extend `PATH` in your `.bashrc` or and don't start Spyder in a console but using a graphical menu or launcher, then Spyder is unable to detect those modifications to `PATH` because it doesn't read your environment variables before launching. By the way, Pycharm also has this problem, as reported here:

https://stackoverflow.com/a/9706384/438386

And I think the same happens in the Jupyter notebook. I mean, I haven't seen any code to read env vars in Jupyter or IPython. But the problem could be alleviated there because you usually start the notebook server in a console, and so their environment is picked by the kernels started by it.

To summarize, I don't know if you could say this is a Spyder problem or people not knowing how env vars work.
